torchvision/prototype/datasets/_builtin/cityscapes.py

```diff
@@ -0,0 +1,201 @@
+            "bicycle": CityscapesClass("bicycle", 33, 18, "vehicle", 7, True, False, (119, 11, 32)),
+            "license plate": CityscapesClass("license plate", -1, -1, "vehicle", 7, False, True, (0, 0, 142)),
+        }
+    )
+
+    def _make_info(self) -> DatasetInfo:
+        name = "cityscapes"
+
+        return CityscapesDatasetInfo(
+            name,
+            categories=list(self.categories_to_details.keys()),
+            homepage="http://www.cityscapes-dataset.com/",
+            valid_options=dict(
+                split=("train", "val", "test", "train_extra"),
+                mode=("fine", "coarse"),
+            ),
+            extra=dict(categories_to_details=self.categories_to_details),
+        )
+
+    _FILES_CHECKSUMS = {
+        "gtCoarse.zip": "3555e09349ed49127053d940eaa66a87a79a175662b329c1a26a58d47e602b5b",
+        "gtFine_trainvaltest.zip": "40461a50097844f400fef147ecaf58b18fd99e14e4917fb7c3bf9c0d87d95884",
+        "leftImg8bit_trainextra.zip": "e41cc14c0c06aad051d52042465d9b8c22bacf6e4c93bb98de273ed7177b7133",
+        "leftImg8bit_trainvaltest.zip": "3ccff9ac1fa1d80a6a064407e589d747ed0657aac7dc495a4403ae1235a37525",
+    }
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+
+        if config.mode == "fine":
+            filenames = ("leftImg8bit_trainvaltest.zip", "gtFine_trainvaltest.zip")
+        else:
+            split_label = "trainextra" if config.split == "train_extra" else "trainvaltest"
+            filenames = (f"leftImg8bit_{split_label}.zip", "gtCoarse.zip")
+
+        resources: List[OnlineResource] = [
+            CityscapesResource(file_name=file_name, sha256=self._FILES_CHECKSUMS[file_name]) for file_name in filenames
+        ]
+        return resources
+
+    def _filter_split_images(self, data: Tuple[str, Any], *, req_split: str) -> bool:
+        path = Path(data[0])
+        split = path.parent.parts[-2]
+        return split == req_split and ".png" == path.suffix
+
+    def _filter_group_targets(self, data: Tuple[str, Any]) -> str:
+        path = Path(data[0])
+        # Taregt path looks like
+        # gtFine/val/frankfurt/frankfurt_000001_066574_gtFine_polygons.json
+        # and we produce the key: frankfurt_000001_066574
+        return path.name.rsplit("_", 1)[0]
+
+    def _prepare_sample(self, data: Tuple[Tuple[str, Any], Any]) -> Dict[str, Any]:
+        (img_path, img_data), target_data = data
+
+        output = dict(image_path=img_path, image=EncodedImage.from_file(img_data))
+        # reorder inside group of targets and setup output dictionary:
+        for path, data in target_data:
+            stem = Path(path).stem
+            for target_type in ["instance", "label", "polygon", "color"]:
+                if target_type in stem:
+                    if target_type == "polygon":
+                        enc_data = json.loads(data.read())
+                    else:
+                        enc_data = EncodedImage.from_file(data)
```

Comment:
Reminder for self: we need an `EncodedSegmentationMask` feature now that we also have a separate `SegementationMask`.